src/main/java/SearchEngine.java

```diff
@@ -0,0 +1,93 @@
+import strategies.impl.AllStrategyImpl;
+import strategies.impl.AnyStrategyImpl;
+import strategies.impl.NoneStrategyImpl;
+import strategies.SearchStrategy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+class SearchEngine {
+
+    private final FileReader fileReader;
+    private final Scanner scanner;
+    private final LineTokenizer lineTokenizer;
+    private List<String> peopleList;
+
+
+    public SearchEngine(FileReader fileReader) {
+        this.fileReader = fileReader;
+        scanner = new Scanner(System.in);
+        lineTokenizer = new LineTokenizer();
+    }
+
+    public void run() throws IOException {
+        peopleList = fileReader.readFile();
+        while (true) {
+            printMenu();
+            String selection = scanner.nextLine();
+            switch (selection) {
+                case "1" -> peopleFinder();
+                case "2" -> printList(peopleList);
+                case "0" -> {
+                    scanner.close();
+                    System.out.println("\nBye!");
+                    return;
+                }
+                default -> System.out.println("\nIncorrect option! Try again.");
+            }
+        }
+    }
+
+    private void printMenu() {
+        System.out.println("""
+                === Menu ===
+                1. Find a person
+                2. Print all people
+                0. Exit""");
+    }
+
+    private void printList(List<String> list) {
```

Comment:
Please position your methods in the order they are used. For example you use peopleFinder() before printList(peopleList); so I expect I will find it first

Reply:
methods are sorted now